src/chronify/time_configs.py

```diff
@@ -159,47 +158,137 @@ class AnnualTimeRange(TimeBaseModel):
     def list_time_columns(self) -> list[str]:
         return [self.time_column]
 
+    def list_time_zone_column(self) -> list[str]:
+        return []
 
-class IndexTimeRange(TimeBaseModel):
-    time_type: Literal[TimeType.INDEX] = TimeType.INDEX
-    start: int
+
+class IndexTimeRangeBase(TimeBaseModel):
+    """Defines a time range in the form of indexes"""
+
+    time_column: str = Field(description="Column in the table that represents index time.")
+    start: int = Field(description="starting index")
     length: int
-    resolution: timedelta
-    time_zone: TimeZone
+    start_timestamp: datetime = Field(
+        description="The timestamp represented by the starting index."
+    )
+    resolution: timedelta = Field(description="The resolution of time represented by the indexes.")
+    time_zone_column: Optional[str] = Field(
+        description="Column in the table that has time zone or offset information.", default=None
+    )
+    time_type: TimeType
+
+    @model_validator(mode="after")
+    def check_time_zone_column(self) -> "IndexTimeRangeBase":
+        tz_col = self.time_zone_column
+        if self.start_timestamp.tzinfo is not None and tz_col is not None:
+            msg = f"{self.start} is tz-aware and {tz_col=} is provided"
+            raise ConflictingInputsError(msg)
+        return self
+
+    def start_time_is_tz_naive(self) -> bool:
+        """Return True if the repreentative timestamps do not have time zones."""
```

Comment:
If you intended to use "representative," it is misspelled. Not sure if you meant that or some flavor of "index."